### PR TITLE
Update indices syntax for BWA index and mem modules

### PR DESCRIPTION
**Pull request by drpatelh**

### CODE REVIEW

File: software/bwa/mem/main.nf [hunk 1]
```diff
@@ -32,13 +31,15 @@ process BWA_MEM {
     def prefix     = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
     def read_group = meta.read_group ? "-R ${meta.read_group}" : ""
     """
+    INDEX=`find -L ./ -name "*.amb" | sed 's/.amb//'`
```
Comment (MaxUlysse) [inline software/bwa/mem/main.nf:34]: I still don't think this is the right way to do it...
And I think you're missing some indices as well

Comment (drpatelh) [reply software/bwa/mem/main.nf:34]: This is just getting the genome prefix automatically. You don't need to list all of the files in the index for that. Just one.

Comment (MaxUlysse) [reply software/bwa/mem/main.nf:34]: why don't you keep the fasta as an input?

Comment (drpatelh) [reply software/bwa/mem/main.nf:34]: This also means you don't need to stage the `fasta` file in the module too and pipelines can accept tar archives / untarred directores containing the indices. You don't even need to know what all of those indices file extensions are as long as the directory contains the index. Much cleaner.

Comment (MaxUlysse) [reply software/bwa/mem/main.nf:34]: yes, but if you don't have the fasta as an input, how will it work?
Also don't you need the fasta index as well?
or do you include that in the index path?

File: software/bwa/mem/main.nf [hunk 2]
```diff
@@ -32,13 +31,15 @@ process BWA_MEM {
...
+
     bwa mem \\
         $options.args \\
         $read_group \\
         -t $task.cpus \\
-        $fasta \\
+        \$INDEX \\
         $reads \\
-        | samtools view $options.args2 -@ $task.cpus -bS -o ${prefix}.bam -
+        | samtools view $options.args2 -@ $task.cpus -bhS -o ${prefix}.bam -
```
Comment (MaxUlysse) [inline software/bwa/mem/main.nf:42]: I think it should be
`samtools $options.args2 --threads $task.cpus -o ${prefix}.bam -`

the `view -bhS` should be part of $options.args2.

File: software/bwa/mem/main.nf (hunk 1 above)
Comment (drpatelh) [reply software/bwa/mem/main.nf:34]: I think the easiest option is to try that command in a directory containing a BWA index and see what it returns 😉

File: software/bwa/mem/main.nf (hunk 2 above)
Comment (drpatelh) [reply software/bwa/mem/main.nf:42]:
```
  -b       output BAM
  -h       include header in SAM output
  -S       ignored (input format is auto-detected)
      --input-fmt-option OPT[=VAL]
               Specify a single input file format option in the form
               of OPTION or OPTION=VALUE
```

Those options are related to the input/output files so should be in the module. What this doesn't cater for are things like CRAM format but it's going to be another level of complexity adding that in here. Maybe save that for another day 😅

**Comment by drpatelh** [comment]
Ping @MaxUlysse.

**Review by MaxUlysse — APPROVED** [review]
ok for now

**Comment by drpatelh** [comment]
Thanks!

**Pull request merged**